### PR TITLE
feat: Add `call‑bind`

### PR DESCRIPTION
**Pull request by ExE-Boss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/call-bind/ts3.9/test/callBind.test.ts [hunk 1]
```diff
@@ -13,7 +13,3 @@ applyBind(() => {}); // $ExpectType (thisArg: unknown, args: readonly []) => voi
 applyBind(Object.prototype.hasOwnProperty);
 // $ExpectType (args: readonly [string | number | symbol]) => boolean || (args: readonly [v: string | number | symbol]) => boolean
 applyBind(Object.prototype.hasOwnProperty, any);
-
-// These are invalid because of `package.json#exports`:
-import callBindIllegal1 = require('call-bind/index'); // $ExpectError
-import callBindIllegal2 = require('call-bind/index.js'); // $ExpectError
```
Comment (ExE-Boss) [inline types/call-bind/ts3.9/test/callBind.test.ts]: @orta @sandersn @weswigham

Why do these lines cause `@types/call‑bind/index.d.ts` to be included in the **TypeScript 3.9** test?

File: types/call-bind/test/callBind.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import callBind = require('call-bind');
+import { apply as applyBind } from 'call-bind';
+
+declare const any: unknown;
```
Comment (ljharb) [inline types/call-bind/test/callBind.test.ts:4]: why this instead of just using `unknown` directly?

**Review by ljharb — APPROVED** [review]

File: types/call-bind/test/callBind.test.ts (hunk 2 above)
Comment (ExE-Boss) [reply types/call-bind/test/callBind.test.ts:4]: Because `unknown` only refers to a type.

**Comment by sandersn** [comment]
@Exe-Boss it's probably because the dependency tracking code expects people to write `require('call-bind')` not `require('call-bind/index')`, and this happens to make a difference in situations typesVersions.

File: types/call-bind/test/callBind.test.ts (hunk 2 above)
Comment (ljharb) [reply types/call-bind/test/callBind.test.ts:4]: gotcha, so it could be named anything then. i was just confused by the name `any`

Comment (sandersn) [reply types/call-bind/test/callBind.test.ts:4]: Agree, `any` is a super confusing name for a value of type `unknown`. Maybe try `unknown` ?

**Review by ljharb — APPROVED** [review]

**Pull request merged**